consul/src/main/java/com/linecorp/armeria/client/consul/ConsulEndpointGroupBuilder.java

```diff
@@ -0,0 +1,117 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.time.Duration;
+
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.consul.ConsulClientBuilder;
+import com.linecorp.armeria.server.consul.ConsulUpdatingListenerBuilder;
+
+/**
+ * A builder class for {@link ConsulEndpointGroup}.
+ */
+public final class ConsulEndpointGroupBuilder extends ConsulClientBuilder {
+    private static final long DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS = 10;
+
+    private final String serviceName;
+    private long registryFetchIntervalSeconds = DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS;
+    private boolean useHealthyEndpoints;
+
+    ConsulEndpointGroupBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+    }
+
+    /**
+     * Sets the interval between fetching registry requests.
+     * If not set, {@value #DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS} is used by default.
+     */
+    public ConsulEndpointGroupBuilder registryFetchInterval(Duration registryFetchInterval) {
+        requireNonNull(registryFetchInterval, "registryFetchInterval");
+        final long seconds = registryFetchInterval.getSeconds();
+        checkArgument(seconds > 0, "registryFetchInterval.getSeconds(): %s (expected: > 0)", seconds);
+        return registryFetchIntervalSeconds(seconds);
+    }
+
+    /**
+     * Sets the interval between fetching registry requests.
+     * If not set {@value #DEFAULT_HEALTH_CHECK_INTERVAL_SECONDS} is used by default.
+     */
+    public ConsulEndpointGroupBuilder registryFetchIntervalSeconds(long registryFetchIntervalSeconds) {
+        checkArgument(registryFetchIntervalSeconds > 0, "registryFetchIntervalSeconds: %s (expected: > 0)",
+                      registryFetchIntervalSeconds);
+        this.registryFetchIntervalSeconds = registryFetchIntervalSeconds;
+        return this;
+    }
+
+    /**
+     * Sets whether to use <a href="https://www.consul.io/api/health.html">Health HTTP endpoint</a>.
+     * Before enabling this feature, make sure that your target endpoints are health-checked by Consul.
+     *
+     * @see ConsulUpdatingListenerBuilder#checkUri(URI)
+     */
+    public ConsulEndpointGroupBuilder useHealthEndpoints(boolean useHealthyEndpoints) {
+        this.useHealthyEndpoints = useHealthyEndpoints;
+        return this;
+    }
+
+    /**
+     * Returns a newly-created {@link ConsulEndpointGroup}.
+     */
+    public ConsulEndpointGroup build() {
+        return new ConsulEndpointGroup(buildClient(), serviceName, registryFetchIntervalSeconds,
+                                       useHealthyEndpoints);
+    }
```

Comment:
Please move this method after all chaining methods for consistency with other builder classes.

Reply:
Ok.